### PR TITLE
Isla cdm code generator workshop

### PR DESCRIPTION
**Pull request by hugohills-regnosys**

### CODE REVIEW

**Review by minesh-s-patel — APPROVED** [review]

File: json-schema/src/main/java/com/regnosys/rosetta/generator/jsonschema/JsonSchemaEnumGenerator.xtend [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			val enumDefinitionContents = enumeration.generateEnumDefinition
+			result.put(enumeration.filename, enumDefinitionContents)
+		}
+
+		result
+	}
+
+	def String generateEnumDefinition(RosettaEnumeration enumeration) '''
+		{
+		  "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (SimonCockx) [inline json-schema/src/main/java/com/regnosys/rosetta/generator/jsonschema/JsonSchemaEnumGenerator.xtend:27]: Maybe better to factor this out into a variable that is reused by all json schema generators, so we can easily change dialects when we want to?

File: json-schema/src/main/java/com/regnosys/rosetta/generator/jsonschema/JsonSchemaTranslator.xtend [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class JsonSchemaTranslator {
+				
+	static def toJsonSchemaBasicType(String typename) {
+		switch typename {
+			case 'string',
+			case 'time',
+			case 'date',
+			case 'dateTime',
+			case 'zonedDateTime',
```
Comment (SimonCockx) [inline json-schema/src/main/java/com/regnosys/rosetta/generator/jsonschema/JsonSchemaTranslator.xtend:13]: Dates and times could be improved. See https://json-schema.org/understanding-json-schema/reference/string#dates-and-times.
(do note the draft version though)

File: json-schema/src/test/resources/basic/com-rosetta-model-lib-meta-Reference.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (SimonCockx) [inline json-schema/src/test/resources/basic/com-rosetta-model-lib-meta-Reference.json:1]: These assertions are empty. Should they have content?

File: json-schema/src/main/java/com/regnosys/rosetta/generator/jsonschema/JsonSchemaMetaFieldGenerator.xtend [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		      "type": "string"
+		    },
+		    "address": {
+		      "$ref": "com-rosetta-model-lib-meta-Reference.schema.json"
+		    }
+		  }
+		}
+	'''
+
+    private def String genMetaFields() '''
```
Comment (SimonCockx) [inline json-schema/src/main/java/com/regnosys/rosetta/generator/jsonschema/JsonSchemaMetaFieldGenerator.xtend:94]: Could these hardcoded schema's be checked in as resources?

**Pull request merged**